core/extensions/impl/crypto_extension.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CRYPTO_EXTENSION_HPP
+#define KAGOME_CRYPTO_EXTENSION_HPP
+
```

Comment:
To say it short, I propose to add include <cstdint> or similar header to hpp files from impl folder, create cpp files and move there function bodies, and add cpp files to project.

The motivation is below:

Here and in other files from extensions/impl directory except for sandboxing_extension.hpp file. My IDE shows error marks on lines where types like uint8_t and others are used, since it cannot see type definitions. 
It is not important right now, since it is not used by itself, but included into a file where type definitions are seen.
However, I have another proposal: maybe you should create cpp files and put function bodies there and same time add them to project file. 
When it comes to real implementation of the functions, we will need these cpp files and we will need put them to project file (CMakeLists.txt). Functions are implemented by different employees and this can lead to merge issues.
Taking this into account I woild ask you to add that cpp files and move function bodies there.
And considering that you have created cpp files, you will need to put include <cstdint> or similar header to hpp files. So maybe you put it there beforehands?

Reply:
Yeah, of course, functions bodies were supposed to be in different .cpp files, I left `std::terminate`-s in headers just because they are not implemented yet and are not going to be called or participate and build processes, but OK, I will create .cpp-s in this PR as well